pkg/controller/operators/catalog/operator.go

```diff
@@ -1240,8 +1246,17 @@ func transitionInstallPlanState(log *logrus.Logger, transitioner installPlanTran
 			out.Status.Phase = v1alpha1.InstallPlanPhaseFailed
 			return out, err
 		}
-		out.Status.SetCondition(v1alpha1.ConditionMet(v1alpha1.InstallPlanInstalled, &now))
-		out.Status.Phase = v1alpha1.InstallPlanPhaseComplete
+		// Loop over one final time to check and see if everything is good.
+		if !out.Status.NeedsRequeue() {
+			log.Info("Status does not need a requeue!")
```

Comment:
I'm not sure if this is leftover from your debugging or you intend to keep it here. I would change it to `log.Debug` because the log message doesn't provide any informative material. Frankly, this is a bit misleading because you requere the InstallPlan, not Status.

Reply:
Leftover code - will remove.